DacDbiInterfaceImpl::IsModuleMapped returns S_FALSE without setting *isModuleMapped

<!--This is just a template - feel free to delete any and all of it and replace as appropriate.-->

### Description

I'm seeing this while debugging a windows dump of a .NET 6 preview app. Essentially, when
querying the mapping mode of a dynamic module in an ASP.NET app, there is no PE image 
for the module, so IsModuleMapped returns S_FALSE without ever setting the out
variable. According to the SAL of the function, the value should be set. It makes sense
that the function should return false in the isModuleMapped out variable in this situation.



### Configuration

debugging a dump collected with the "createdump" app of a .NET 6 preview 7.21379.14 running an x86 ASP.NET app on Windows.

### Regression?

No. This is a new API

### Other information

This can be worked-around by setting the value of the output variable to false before calling the API, and by checking for S_OK instead of hr > 0 for success.

The fix should be rather simple. I can supply it. I also have a dump that exercises this code if it is helpful.

I have a fix for this, build cmd is giving me the following errors when I try to build:

```
E:\dotnetruntime\.dotnet\sdk\6.0.100-rc.1.21430.12\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error NU5118: File 'E:\dotnetruntime\artifacts\bin\coreclr\windows.x64.Debug\PDB\crossgen2.pdb' is not added because the package already contains file 'runtimes\win-x64\lib\net7.0\..\..\..\..\tools\crossgen2.pdb' [E:\dotnetruntime\src\installer\pkg\sfx\Microsoft.NETCore.App\Microsoft.NETCore.App.Crossgen2.sfxproj]
E:\dotnetruntime\.dotnet\sdk\6.0.100-rc.1.21430.12\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error NU5118: File 'E:\dotnetruntime\artifacts\bin\coreclr\windows.x64.Debug\PDB\ILCompiler.DependencyAnalysisFramework.pdb' is not added because the package already contains file 'runtimes\win-x64\lib\net7.0\..\..\..\..\tools\ILCompiler.DependencyAnalysisFramework.pdb' [E:\dotnetruntime\src\installer\pkg\sfx\Microsoft.NETCore.App\Microsoft.NETCore.App.Crossgen2.sfxproj]
E:\dotnetruntime\.dotnet\sdk\6.0.100-rc.1.21430.12\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error NU5118: File 'E:\dotnetruntime\artifacts\bin\coreclr\windows.x64.Debug\PDB\ILCompiler.Diagnostics.pdb' is not added because the package already contains file 'runtimes\win-x64\lib\net7.0\..\..\..\..\tools\ILCompiler.Diagnostics.pdb' [E:\dotnetruntime\src\installer\pkg\sfx\Microsoft.NETCore.App\Microsoft.NETCore.App.Crossgen2.sfxproj]
E:\dotnetruntime\.dotnet\sdk\6.0.100-rc.1.21430.12\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error NU5118: File 'E:\dotnetruntime\artifacts\bin\coreclr\windows.x64.Debug\PDB\ILCompiler.ReadyToRun.pdb' is not added because the package already contains file 'runtimes\win-x64\lib\net7.0\..\..\..\..\tools\ILCompiler.ReadyToRun.pdb' [E:\dotnetruntime\src\installer\pkg\sfx\Microsoft.NETCore.App\Microsoft.NETCore.App.Crossgen2.sfxproj]
E:\dotnetruntime\.dotnet\sdk\6.0.100-rc.1.21430.12\Sdks\NuGet.Build.Tasks.Pack\build\NuGet.Build.Tasks.Pack.targets(221,5): error NU5118: File 'E:\dotnetruntime\artifacts\bin\coreclr\windows.x64.Debug\PDB\ILCompiler.TypeSystem.ReadyToRun.pdb' is not added because the package already contains file 'runtimes\win-x64\lib\net7.0\..\..\..\..\tools\ILCompiler.TypeSystem.ReadyToRun.pdb' [E:\dotnetruntime\src\installer\pkg\sfx\Microsoft.NETCore.App\Microsoft.NETCore.App.Crossgen2.sfxproj]
```
Any ideas waht I should do?

--- a/src/coreclr/debug/di/module.cpp
+++ b/src/coreclr/debug/di/module.cpp
@@ -2755,6 +2755,7 @@ HRESULT CordbModule::IsMappedLayout(BOOL *isMapped)
     FAIL_IF_NEUTERED(this);
 
     HRESULT hr = S_OK;
+    *isMapped = FALSE;
     CordbProcess *pProcess = GetProcess();
 
     ATT_REQUIRE_STOPPED_MAY_FAIL(pProcess);

--- a/src/coreclr/inc/cordebug.idl
+++ b/src/coreclr/inc/cordebug.idl
@@ -5219,7 +5219,12 @@ interface ICorDebugModule4 : IUnknown
                       format while FALSE represents flat format.
      * Return Value:
      *     S_OK in successful case.
+     *     S_FALSE if they layout could not be determined.
      * Notes:
+     *     The pIsMapped value set in pIsMapped should only be interpreted as valid
+     *     when this function returns S_OK. All other return values (including
+     *     S_FALSE) indicate that the layout could not be determined and pIsMapped
+     *     should be ignored.
      */
     HRESULT IsMappedLayout([out] BOOL *pIsMapped);
 }

